### PR TITLE
Medic outgoing registration dao service

### PR DESCRIPTION
**Pull request by PatrickWaweru**
Added the DAO, service, service implementation and service config

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	public void purgeRecord(MedicOutgoingRegistration record) {
+		getSession().delete(record);
+	}
+	
+	public void voidRecord(Integer id) {
+		MedicOutgoingRegistration record = (MedicOutgoingRegistration) getSession().get(MedicOutgoingRegistration.class, id);
+		record.setVoided(true);
+	}
+	
+	public List<MedicOutgoingRegistration> getRecordsByStatus(Integer status) {
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java:76]: Can we enforce an ordered result i.e first records to come at the top?

File: api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	private DbSession getSession() {
+		return sessionFactory.getCurrentSession();
+	}
+	
+	public MedicOutgoingRegistration getRecordByUuid(String uuid) {
+		return (MedicOutgoingRegistration) getSession().createCriteria(MedicOutgoingRegistration.class)
+		        .add(Restrictions.eq("uuid", uuid)).uniqueResult();
+	}
+	
+	public MedicOutgoingRegistration saveRecord(MedicOutgoingRegistration record) {
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java:38]: We are used to saveOrUpdate methods in OpenMRS. May be you can change the name to follow the convention.

File: api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	 */
+	@Transactional(readOnly = true)
+	public MedicOutgoingRegistration getRecordByKemrRef(String kemrRef);
+	
+	/**
+	 * Deletes a record from the database.
+	 * 
+	 * @param record the record to delete.
+	 */
+	@Authorized(AfyastatConfig.MODULE_PRIVILEGE)
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java:101]: Have you tested how this privilege works? 99% usage of the method will be through scheduler and a few times by users. Defining the privilege for the method would require that who/what invokes the method has the privilege. Please test it out and make any modifications

**Review by ojwanganto — CHANGES_REQUESTED** [review]
This is good work, Patrick! Please address the few comments so that we review again.

File: api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java (hunk 1 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java:79]: Done

File: api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java (hunk 2 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/afyastat/api/db/MedicOutgoingRegistrationDao.java]: Done

File: api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java (hunk 3 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java:100]: Tested and confirmed working for the schedulers.

File: api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
 public interface MedicOutgoingRegistrationService extends OpenmrsService {
 	
 	/**
-	 * Returns an item by uuid. It can be called by any authenticated user. It is fetched in read
+	 * Returns an record by uuid. It can be called by any authenticated user. It is fetched in read
```
Comment (makombe) [inline api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java:29]: Should read 'Returns a record' not 'an record'

Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/afyastat/api/service/MedicOutgoingRegistrationService.java]: Done

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**